testing-guidance/workflow-tests/scripting/lua-scripting-tests.md

```diff
@@ -0,0 +1,293 @@
+
+## Area: Creating a Lua skeleton script file
+
+### Project Requirements
+Any project can be used.
+
+**Product:** 
+Skeleton Lua script assigned to an entity and printing simple messages to console when entering/exiting Game Mode.
+
+**Suggested Time Box:** 
+20 minutes
+
+| Workflow | Requests | Things to Watch For |
+|----------|----------|---------------------|
+| Create a Lua skeleton script | <ol><li>Create a new .lua file by selecting *File → New* in Lua Editor, make sure to save it in a path accessible to Editor (for example *<project path\>\\Assets\\*</li><li>Add a skeleton Lua script to the file, as seen here: https://www.o3de.org/docs/user-guide/scripting/lua/basic-lua-script/#skeleton-script - remember to replace *BaseScript* in the file to the name of your lua file if it is different (you can do this easily using *Find* tool and *Replace All* action)</li><li>Replace `-- Activation Code` and `-- Deactivation Code` lines with `Debug.Log(tostring("Hi!"));` and `Debug.Log(tostring("Bye!"));` respectively</li><li>Save the .lua file (for example *File → Save* or *Ctrl+S*)</li></ol>| <ul><li>Basic script can be created and saved</li><li>After saving *Compilation Succesful - <path\>* is printed to Log and no errors occur</li></ul>|
+| Create an entity using the created skeleton Lua script | <ol><li>In Editor, create a new entity and add a *Lua Script* component to it</li><li>Select *Lua Script* component on your new entity, click the *Pick Lua Script* button and choose your skeleton script</li><li>Enter and exit Game Mode (using for example *Ctrl+G*) while paying attention to the Editor's Console</li></ol> | <ul><li>*Lua Script* component can be assigned to an entity</li><li>Lua script can be assigned to *Lua Script* component</li><li>*Hi!* is printed to Console on entering the Game Mode and *Bye!* on exiting it</li><li>Editor remains stable</li><li>Final working script can be found below this table</li></ul> |
+
+**"Area: Creating a Lua skeleton script file" script:**
+```
+-- BaseScript.lua 
+
+local BaseScript = 
+{
+    Properties =
+    {
+        -- Property definitions
+    }
+}
+
+function BaseScript:OnActivate()
+     Debug.Log(tostring("Hi!"));
+end
+
+function BaseScript:OnDeactivate()
+     Debug.Log(tostring("Bye!"));
+end
+
+return BaseScript
+```
```

Comment:
Let's put the code here in a "collapsed section".

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections

Reply:
Adjusted this and other similar code sections as described.